### ISSUE TITLE
Visits not inserted in log_visit and no errors triggered (unknown reason)

### ISSUE THREAD

**Issue by mattab** [matomo-org/matomo#13755]
Reported in https://forum.matomo.org/t/no-longer-logging-vists-piwik-php-called/29700/3

We can see in the tracker debug output that the Visit insertion seems to have occured, but actually failed:

```

DEBUG: array (
DEBUG:   'idvisitor' => '1db1ff046f15737f',
DEBUG:   'config_id' => '5c6948980cd42fd4',
DEBUG:   'location_ip' => 'xxx.xxx.xx.0',
DEBUG:   'idsite' => 1,
DEBUG:   'visit_first_action_time' => '2018-09-05 19:48:42',
[....]
DEBUG: )
DEBUG: Executing Piwik\Plugins\CoreHome\Tracker\VisitRequestProcessor::recordLogs()...
DEBUG: Executing Piwik\Plugins\Actions\Tracker\ActionsRequestProcessor::recordLogs()...
DEBUG: Inserted new action:
DEBUG: array (
DEBUG:   'idvisit' => 0,
DEBUG:   'idsite' => 1,
DEBUG:   'idvisitor' => '1db1ff046f15737f',
```

Here we see that ` 'idvisit' => 0,` so it seems the log_visit insert went wrong? it is reproducible for this user and hasn't worked since months ago. Maybe the problem is a corrupted log_visit table but it could be anything really..

Suggested steps:

* In the tracker debug log, let's add at the end the output of `SHOW WARNINGS` query so we can maybe learn about some hidden problems
* maybe we need to double check in the code that we catch all possible errors/warnings from MySQL inserts during tracker
*  in the tracker code could we add a check if idvisit == 0 then we can exit with an error and not record the data? this way it's more likely these problems will be detected.
* what do you think?

**Comment by tsteur** [comment]
Show warnings would need to be executed after each query or not? Not just at the end...

**Comment by tsteur** [comment]
@tyler71 before we invest any time in this please let us know your MySQL version and `show create table `matomo_log_visit` (or `piwik_log_visit`).

Does it record this page view when you open the URL? `https://$DOMAIN/piwik.php?rec=1&idsite=1&action_name=foo`
Ideally you would give us view access to your system in order to help you.

**Comment by tyler71** [comment]
`show create table`
https://gist.github.com/tyler71/ad318f40bb00417ea1f1d2a19c101ee4

The page view is not recorded when opened with browsers.
`wget` shows the pixel image loads

> Ideally you would give us view access to your system in order to help you.
Like a account in my matomo instance?

**Comment by tsteur** [comment]
Do other things work in your Matomo instance such as creating a scheduled report? Updating password, or making any change to settings in Admin => General Settings?

I'm thinking on this instance maybe `autocommit` is disabled or so.

**Comment by tyler71** [comment]
I updated my password and it worked fine

Attempting to navigate around the site is now resulting in a error
```
Oops… there was a problem during the request. Maybe the server had a temporary issue, or maybe you requested a report with too much data. Please try again. If this error occurs repeatedly please contact your Matomo administrator for assistance. 
```
Nginx Error logs reveal this:
```
sudo tail -fn0 /var/log/nginx/error.log
2019/04/04 13:29:50 [error] 521#521: *34435 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in /html/piwik/libs/Zend/Db/Statement/Mysqli.php on line 304" while reading response header from upstream, client: xx.xx.x.xx, server: my.foobardomain.com, request: "POST /index.php?date=2019-03-29,2019-04-04&module=Live&action=widget&widget=1&disableLink=0&idSite=4&period=range HTTP/1.1", upstream: "fastcgi://unix:/var/run/php7-fpm.services.sock:", host: "my.foobardomain.com"
2019/04/04 13:29:50 [error] 528#528: *34427 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in /html/piwik/libs/Zend/Db/Statement/Mysqli.php on line 304" while reading response header from upstream, client: xx.xx.x.xx, server: my.foobardomain.com, request: "POST /index.php?period=range&idSite=4&segment=&date=last3&format=json&showRawMetrics=1&module=API&method=Live.getLastVisitsDetails&filter_limit=100&showColumns=latitude%2Clongitude%2Cactions%2ClastActionTimestamp%2CvisitLocalTime%2Ccity%2Ccountry%2CcountryCode%2CreferrerType%2CreferrerName%2CreferrerTypeName%2CbrowserIcon%2CoperatingSystemIcon%2CdeviceType%2CdeviceModel%2CcountryFlag%2CidVisit%2CactionDetails%2CcontinentCode%2Cactions%2Csearches%2CgoalConversions%2CvisitorId%2CuserId&minTimestamp=0 HTTP/1.1", upstream: "fastcgi://unix:/var/run/php7-fpm.services.sock:", host: "my.foobardomain.com"
```
Memory leak?

**Comment by tsteur** [comment]
You definitely want to increase your memory limit to say 256MB or even 512MB. 128MB can be quite low.

**Comment by tyler71** [comment]
> Attempting to navigate around the site is now resulting in a error in specific areas (Visits in real time, visits logs, etc)

Changed to 256MB resolved the above error

Unfortunately, no visits are recording still

**Comment by tsteur** [comment]
>> Ideally you would give us view access to your system in order to help you.

> Like a account in my matomo instance?

We would ideally need a login for your Matomo instance, plus access to the code on the server so we can investigate the issue.

**Comment by mattab** [comment]
Hi @tyler71 are you still experiencing this issue? If so, be great if you can reach out to us via: http://matomo.org/contact/
and we'll follow up and investigate with you. Thanks!

**Comment by genachka** [comment]
@mattab not sure if I'm having the same issue, but after upgrading to 3.11.0 from 3.9 2 days ago none of my new visits are being recorded for any of my sites. I'm running in docker (via crazymax/matomo:3.11.0 container) and traefik (via traefik 1.7.14 container) and mariadb (via mariadb:10.4.7 container). Console log shows that the requests come in but it's as if they aren't being stored in database. I'm using a cron sidecar and it reports no issues doing archive. I tried to manually update the database on console as well as archive (always now says no new visits) and no errors reported.

**Comment by tsteur** [comment]
I recommend you check in `matomo_log_visit` table (name depends on your table prefix) whether there are actually recent visits coming in (sort by idvisit and keep refereshing the table). If not, you will want to debug the tracker: https://developer.matomo.org/api-reference/tracking-api#debugging-the-tracker

**Comment by genachka** [comment]
@tsteur thanks for your suggestions. Looks like last written was on the 13th right before I upgraded.

**Comment by genachka** [comment]
1. I left the database container at 10.4.7 but restored the database I had with 3.9.1 and reverted matomo container to 3.9.1 and the cron sidecar to 3.9.1 and it started writing to the database.
2. I then upgraded both containers to 3.11.0 and same results (nothing new being written to db).
3. I left the database container at 10.4.7 but restored the database I had with 3.9.1 and set matomo container to 3.10.0 and the cron sidecar to 3.10.0 and it started writing to the database too.
4. I then upgraded both containers to 3.11.0 and same results (nothing new being written to db).
So appears to be something with the 3.11.0 upgrade... :( Going to 3.10.0 for now.

**Comment by mattab** [comment]
@genachka can you follow instructions at https://developer.matomo.org/api-reference/tracking-api#debugging-the-tracker and see what is the output message in debug mode?

**Comment by genachka** [comment]
@mattab just to follow up here, I didn't have the chance to do as you suggested so I just left it running on the reverted 3.10.0 until 3.12.0 was released. I upgraded to 3.12.0 without the issues. Maybe it was something with the crazymax/matomo:3.11.0 container or something in the Matomo 3.11.0 code. But all good now.

Running without issues for almost a week on with the following containers:
* traefik:1.7.19
* crazymax/matomo:3.12.0
* Matomo:3.12.0
* MariaDB:10.4.8
* PHP:7.3.11
